### PR TITLE
Extended XMTZC05HM decoder

### PR DESCRIPTION
**Pull request by DigiH**
- Extension of model and property condtions by @h2zero to allow for the extension of the XMTZC05HM decoder
- Pulled back _XMTZC05HM_json_props into XMTZC05HM_json.h
- Renamed "model" to more informative "Mi_Body_Scale_2" and amended description in the docs 
- Implemented 'stabilised' only decoding for XMTZC05HM
- Recognises 'stabilised' 2 (scale display active) and 'stabilised' a (scale display deactivated) to guarantee published 'stabilised' reading without the need for very short TimeBtwRead
- New 'weighing_mode' attribute to indicate if the weighing was of a person or a small object (if this option is activated in the scale's settings). This allows filtering to only record person weighings in HA or openHAB
- Only publish 'impedance' attribute if the weighing of a person was done barefoot and the impedance value could be recorded correctl
- added all 12 possible variations in tests for now

I suspect that the 1st version of the Mi Body Composition Scale (XMTZCO2HM) is also being recognised by this decoder, and should therefor be mentioned in the docs and published model_id, but without confirmation from XMTZCO2HM owners this is just an assumption.

## Checklist:
  - [x] The pull request is done against the latest development branch
  - [x] Only one feature/fix was added per PR and the code change compiles without warnings
  - [x] I accept the [DCO](https://github.com/theengs/decoder/blob/development/docs/participate/development.md#developer-certificate-of-origin).

### CODE REVIEW

File: tests/BLE/test_ble.cpp [hunk 1]
```diff
@@ -69,8 +69,18 @@ const char* expected_mfg[] = {
 const char* expected_uuid[] = {
     "{\"brand\":\"Xiaomi\",\"model\":\"Miband\",\"model_id\":\"MiBand\",\"steps\":7842}",
     "{\"brand\":\"Xiaomi\",\"model\":\"Miscale_v1\",\"model_id\":\"XMTZC04HM\",\"unit\":\"kg\",\"weight\":61.75}",
-    "{\"brand\":\"Xiaomi\",\"model\":\"Miscale_v2\",\"model_id\":\"XMTZC05HM\",\"unit\":\"kg\",\"weight\":72.45,\"impedance\":503}",
-    "{\"brand\":\"Mokosmart\",\"model\":\"Beacon\",\"model_id\":\"Mokobeacon\",\"batt\":100,\"x_axis\":-24576,\"y_axis\":-3841,\"z_axis\":-8189}",
```
Comment (h2zero) [inline tests/BLE/test_ble.cpp:73]: This line should be replaced on line 84 or the tests will fail.

**Comment by DigiH** [comment]
I shouldn't hurry with things like that ;) thanks

**Review by h2zero — APPROVED** [review]
LGTM, thanks!

**Comment by 1technophile** [comment]
Thanks!

**Pull request merged**